### PR TITLE
Fix for ES7 : Specifying types is deprecated.

### PR DESCRIPTION
**Pull request by oxarbitrage**
Reported in telegram by @HarukaMa . Elasticsearch plugin is not working in ES7 as `_type` is deprecated. However this needs to be there for ES6(and probably lower versions).

This pull send a query to the server, get the version string and send the type accordingly in both plugins(`elasticsearch` and `es_objects`)

### CODE REVIEW

**Comment by abitmore** [comment]
I'm not sure whether this PR fixed a problem, but my node is running ES7 fine with default `config.ini`. The issue @HarukaMa reported was an error when this parameter is set to `2`:
```
# Mode of operation: only_save(0), only_query(1), all(2) - Default: 0
# elasticsearch-mode = 
```

**Comment by oxarbitrage** [comment]
He later reported the issue was happening for him also in only_save(`0`) iirc.

**Comment by oxarbitrage** [comment]
In conversations in the dev channel yesterday he fixed by removing the `_type` from the queries just as we are doing here except that we need to keep functionality for ES6. I am positive that it will not work with types removed on ES6.
Not sure why your ES7 is working without this change while his not. A different minor version ?

Will try ES7 myself.

File: libraries/plugins/es_objects/es_objects.cpp [hunk 1]
```diff
@@ -403,6 +406,10 @@ void es_objects_plugin::plugin_startup()
    if(!graphene::utilities::checkES(es))
       FC_THROW_EXCEPTION(fc::exception, "ES database is not up in url ${url}", ("url", my->_es_objects_elasticsearch_url));
    ilog("elasticsearch OBJECTS: plugin_startup() begin");
+
+   const auto es_version = graphene::utilities::getVersion(es);
+   if(std::stoi(es_version.substr(0,1)) >= 7)
```
Comment (oxarbitrage) [inline libraries/plugins/es_objects/es_objects.cpp]: This is wrong, will fail at ES10. Will change.

Comment (oxarbitrage) [reply libraries/plugins/es_objects/es_objects.cpp]: should be better with https://github.com/bitshares/bitshares-core/pull/1997/commits/ae067f0999bd209444bde86dd3b8d481e10ff904

**Comment by oxarbitrage** [comment]
I found the details of the error @HarukaMa was getting:
```
Rejecting mapping update to [bitshares-2015-10] as the final mapping would have more than 1 type: [_doc, data]
```
Version string is: `7.3.1`

@abitmore can you post your version? `curl -XGET http://localhost:9200/` will do it.

**Comment by abitmore** [comment]
My node is `7.3.0`.

**Comment by HarukaMa** [comment]
Now thinking that, I think it is possible that is because I have templates to adjust the parameters (like suppressing refreshing during initial index, changing replica counts and some settings optimized for spinning disks), and the template itself may have implied _doc type although it’s not explicitly set.

Anyways, the type field itself is indeed deprecated.

iPhoneから送信

2019/09/22 8:32、Abit <notifications@github.com>のメール:

**Comment by oxarbitrage** [comment]
This by now is not needed. i downloaded and tested elasticsearch 7.4.0 (last stable version) without any issue. The pull can be handy in the future but not needed now.

**Comment by abitmore** [comment]
This is still needed. In ES8 the warning will become an error.

**Comment by abitmore** [comment]
Those who are interested, please check https://github.com/bitshares/bitshares-core/issues/2240#issuecomment-997296211 for more info.

**Pull request merged**